### PR TITLE
Add build image for Debian 11 (Bullseye)

### PR DESCRIPTION
**Pull request by ridwanmsharif**
This change adds the commands for a Debian 11 image. Uses Debian 10 as a
template.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -16,6 +16,24 @@
 # or DOCKER_BUILDKIT=1 docker build -o /tmp/out . --target=buster
 # Generated tarball(s) will end up in /tmp/out
 
+FROM debian:bullseye AS bullseye-build
+
+# TODO: Factor out the common code without rerunning apt-get on every build.
```
Comment (jefferbrecht) [inline Dockerfile:21]: Minor, if we're keeping this TODO around it's probably fine to just leave it on the first entry and remove the duplicated one from the buster section.

**Review by jefferbrecht — APPROVED** [review]

**Comment by jefferbrecht** [comment]
Kokoro should be working now. Not sure if there's a way to re-trigger the actions without committing something.

File: Dockerfile (hunk 1 above)
Comment (ridwanmsharif) [reply Dockerfile]: Done.

File: Dockerfile [hunk 2]
```diff
@@ -16,9 +16,9 @@
 # or DOCKER_BUILDKIT=1 docker build -o /tmp/out . --target=buster
 # Generated tarball(s) will end up in /tmp/out
 
-FROM debian:buster AS buster-build
+FROM debian:bullseye AS bullseye-build
 
-# TODO: Factor out the common code without rerunning apt-get on every build.
+# TODO: Factor out the common code without rerunning apt-get on every debian build.
```
Comment (quentinmit) [inline Dockerfile:21]: I believe this comment applies to both Debian and Ubuntu

**Review by quentinmit — APPROVED** [review]

**Pull request merged**